### DISCUSSION TITLE
Accepting contributions?

### DISCUSSION THREAD

**Discussion by AnthonyMichaelTDM** [RustyRoad/RustyRoad#18] [Ideas]
I saw this project, and it piqued my interest, though it seems like you're in the process of splitting it into separate repos under the [RustyRoad](https://github.com/RustyRoad) org, so I am curious which repos are the ones being currently developed?

I'd probably start with adding a CI pipeline, and fixing any slippy lints or minor bugs said CI brings up (if any), then move on to some of the TODO items you have scattered around.

I've worked with a [similar project](https://github.com/Wulf/create-rust-app) in the past, but that one is falling victim to technical debt, and I think this project has the potential to hit a sorely under-developed niche in the rust ecosystem.

Another question, I see diesel mentioned in the README, but I don't see it used much in the code, what's up with that? If it's just to have an ORM, there is the wonderful [SeaORM](https://github.com/SeaQL/sea-orm) (built off of sqlx, which you already are using) that may be a valid contender to replace diesel's role, but honestly I'm not sure whether an ORM would be of much value considering existing the features of the CLI.

**Comment by rileyseaburg** [comment]
@AnthonyMichaelTDM You can definitely make a PR.

I'm in the process of creating a full stack solution including all the various repos under [RustyRoad](https://github.com/RustyRoad) org.

I decided to stabilize the migration API before I make the transition because the framework is incomplete without it. 

I've been having issues mapping the various db objects to the CLI. 
I see that SeaORM is a pretty great tool and it reminds me a lot of diesel. 
I decided not to use the Diesel as the ORM because I could not ship the migration CLI in binary.
For this reason, I decided to implement a migration CLI manually and have it generate and parse raw SQL with SQLX. 

I see that SeaORM has some [solutions](https://github.com/SeaQL/sea-orm/blob/master/examples/actix_example/migration/src/main.rs) for that built-in.

I'm going to test this in a feature branch. 

I'll keep this open and we can make adding this a feature to work on.

**Comment by jondot** [comment]
@RileySeaburg I'm going to start working on a Rails for Rust project. I have [previous experience with these things](https://hyperstackjs.io).

My main blocker was a stable, fast (for development, not runtime) ORM. Diesel has been really challenging, while SeaORM finally is being a viable option -- which is why I'm greenlighting this to myself.

SeaORM has still some cruft with regards to relationship set up, but it is what it is.

If you want to collab, I'd be happy

**Comment by rileyseaburg** [comment]
I can invite @jondot  and @AnthonyMichaelTDM as collaborators on this project. 

Just be advised the repo will be moving to https://github.com/rustyroad.
I'm going to transfer ownership of this repo so just keep that in mind. 

Currently, I am stabilizing the migration API and the route generation API. 

If you guys want to get started on implementing SeaOrm it would be super helpful. 

My next step is to integrate [Leptos](https://github.com/leptos-rs/leptos)

**Comment by rileyseaburg** [comment]
didn't mean to close this

**Comment by rileyseaburg** [comment]
Invites sent to both of you

**Comment by AnthonyMichaelTDM** [comment]
Awesome, thanks

**Comment by rileyseaburg** [comment]
We are going to add SeaOrm as an option in the cli, because not everybody likes to use an ORM

**Comment by AnthonyMichaelTDM** [comment]
Sounds Like a good idea
